gym_sapientino/core/states.py

```diff
@@ -43,13 +45,25 @@ def __init__(self, config: "SapientinoConfiguration"):
         self.score = 0
         self._grid = self.config.grid
         self._grid.reset()
-        self._robots: List[Robot] = [
-            Robot(config, 1 + 2 * i, 2, 0.0, 90.0, i) for i in range(config.nb_robots)
-        ]
+        self._robots: List[Robot] = self._make_robots(config)
         self._last_commands: List[COMMAND_TYPES] = [
             ac.action_type.NOP for ac in self.config.agent_configs
         ]
 
+    @staticmethod
+    def _make_robots(config: SapientinoConfiguration) -> List[Robot]:
+        """Instantiate the robot objects."""
```

Comment:
This initializes the robots based on initial states too. It should be backward compatible with existing code